li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -437,7 +437,10 @@ public long getRecordsSkipped() {
     TopicPartition tp = new TopicPartition(consumerRecord.topic(), consumerRecord.partition());
     ConsumerRecord<K, V> handledRecord = null;
     K key = _keyDeserializer.deserialize(tp.topic(), consumerRecord.key());
-    byte[] valueBytes = parseAndMaybeTrackRecord(tp, consumerRecord.offset(), consumerRecord.value());
+    // Create a new copy of the headers
+    Headers headers = new RecordHeaders(consumerRecord.headers());
```

Comment:
nit: Why is creating a copy of the headers necessary here?

Reply:
I agree it is somehow redundant. But that's from our old ComsumerRecordProceeor. Lets discuss in the meeting if it has some other purpose to have such a copy

Reply:
After discussion, let's leave it not changed

Reply:
sg!